pkg/systemd/register.go

```diff
@@ -0,0 +1,34 @@
+package systemd
+
+import (
+	"encoding/hex"
+	"github.com/godbus/dbus"
+)
+
+var conn *dbus.Conn
+
+// RegisterMachine with systemd on the host system
+func RegisterMachine(name string, id string, pid int, root_directory string) error {
+	var (
+		av  []byte
+		err error
+	)
+	if !SdBooted() {
```

Comment:
should we move this check outside of the library to make explicit about the fact that we don't run this when systemd is not running?